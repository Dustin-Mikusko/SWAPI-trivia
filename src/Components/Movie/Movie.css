article {
  margin: 3% 1% 1% 1%;
  background-color: white;
  width: 320px;
  height: 590px;
  border: 10px solid white;
  position: relative;
  color: #000;
  transition: all .3s ease-in-out;
}

article:hover {
  transform: scale(1.03);
  box-shadow: 2px 2px 30px 15px rgba(255, 0, 0, 0.776);
}

h1 {
  position: absolute;
  color: white;
  top: 20px;
  left: 56px;
  font-size: 45px;
  font-family: 'Cormorant Garamond', serif;
  text-shadow: 2px 1.5px 1px black;
}

.image {
  background-image: url('../../images/darth.jpg');
  background-size: cover;
  width: 100%;
  height: 472px;
  margin-bottom: 18px;
}

h3 {
  margin: 0px 0px 10px 0px;
}

.characters-button-container {
  width: 100%;
  height: 28px;
  display: flex;
  justify-content: center;
  text-align: center;
}

.characters {
  width: 100%;

  font-size: 1.25em;
  border-radius: 5px;
  background-color: #E8C539;
  margin: auto;
  font-weight: 600;
  text-decoration: none;
  color: #000;
}

.characters:hover {
  cursor: pointer;
  font-weight: 800;
}

.characters:active {
  transform: scale(.97);
}

span {
  font-weight: 400;
}
