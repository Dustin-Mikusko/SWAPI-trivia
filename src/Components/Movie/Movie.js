import React, { Component } from 'react';
import './Movie.css';
import poster from '../../images/movie.jpeg';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Movie = ({ title, episode, releaseDate, showMovieCharacters }) => {
  return (
    <article>
      <div className="image">
      <h1>EPISODE {episode}</h1>
      </div>
      <h3>Title: <span>{title}</span></h3>
      <h3>Release Date: <span>{releaseDate}</span></h3>
      <div className="characters-button-container">
        <Link onClick={showMovieCharacters} to={`/movies/${episode}`} className="characters" >View Characters
        </Link>
      </div>
    </article>
  )

  }





export default Movie;

Movie.propTypes = {
  title: PropTypes.string,
  episode: PropTypes.string,
  releaseDate: PropTypes.string,
  showMovieCharacters: PropTypes.func
}
