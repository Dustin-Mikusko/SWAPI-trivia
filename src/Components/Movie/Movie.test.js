import React from 'react';
import { shallow } from 'enzyme';
import Movie from './Movie';

describe('Movie', () => {
  const mockMovie = {
    key: 1,
    title: "Revenge of the Sith",
    episode: "4",
    releaseDate: "2002",
    image: "url.to.movie",
    openingCredits: "WeeWooWeeWoo",
    characters: [{name: 'Tom Brady'}],
    showMovieCharacters: jest.fn()
  };

  let wrapper = shallow(<Movie 
    key={mockMovie.key}
    title={mockMovie.title}
    episode={mockMovie.episode}
    releaseDate={mockMovie.releaseDate}
    image={mockMovie.image}
    openingCredits={mockMovie.openingCredits}
    characters={mockMovie.characters}
  />)

  it('should render the component with the correct data to match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call checkMovies when the Link tag is clicked', () => {
    const mockShowMovieCharacters = jest.fn();
    wrapper = shallow(<Movie 
      key={mockMovie.key}
      title={mockMovie.title}
      episode={mockMovie.episode}
      releaseDate={mockMovie.releaseDate}
      image={mockMovie.image}
      openingCredits={mockMovie.openingCredits}
      characters={mockMovie.characters}
      showMovieCharacters={mockShowMovieCharacters}
    />)

    wrapper.find('Link').simulate('click');

    expect(mockShowMovieCharacters).toHaveBeenCalled();
  })
})
