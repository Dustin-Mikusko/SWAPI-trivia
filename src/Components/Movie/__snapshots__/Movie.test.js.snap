// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render the component with the correct data to match snapshot 1`] = `
<article>
  <div
    className="image"
  >
    <h1>
      EPISODE 
      4
    </h1>
  </div>
  <h3>
    Title: 
    <span>
      Revenge of the Sith
    </span>
  </h3>
  <h3>
    Release Date: 
    <span>
      2002
    </span>
  </h3>
  <Link
    className="characters"
    to="/movies/4"
  >
    View Characters
  </Link>
</article>
`;
